File: similar-triangles-verification.cpp
#include <iostream>
#include <cstdlib>


void printMemory(int memory[], int size)
{
	//std::cout << "print memory" << std::endl;
	for (int i = 0; i < size; i++)
		std::cout << memory[i] << " ";
	std::cout << std::endl;
}

void bubbleSort(int dataIndex, int memory[])
{
	//printMemory(memory, 6);
	int R_1 = 0;
	int R_2 = 0;
	int dataCounter = dataIndex;
	bool flag = true;
	while (flag == true)
	{
		dataCounter = dataIndex;
		flag = false;
		while (dataCounter < dataIndex + 2)
		{
			R_1= memory[dataCounter];
			R_2= memory[dataCounter + 1];
			if (memory[dataCounter] < memory[dataCounter + 1])
			{
				memory[dataCounter + 1] = R_1;
				memory[dataCounter] = R_2;
				flag = true;
			}
			dataCounter++;
		}
	}
}

int main(int argc, char *argv[])
{
	std::cout << "---start similar triangles verfication---" << std::endl;
	// memory and parameter initialisation
	int memory[6] = {3, 3, 4, 4, 6, 8};
	int index = 0;
	int R_1 = 0;
	int R_2 = 0;
	int product_1 = 0;
	int product_2 = 0;
	if (argc != 7)
	{
		std::cerr << "Invalid number of input arguments\nPlease input the side lengths of 2 triangles" << std::endl;
		return 1;
	}
	else
	{
		for (int i = 1; i < argc; i++ )
		{
			memory[i-1] = atoi(argv[i]);
		}
	
	}
	std::cout << "initial memory" << std::endl;
	printMemory(memory, 6);

	while (index < 2)
	{
		if (index == 1)
		{
			bubbleSort(3, memory);
			std::cout << "memory after second sort" << std::endl;
			printMemory(memory, 6);
		}
		else
		{
			bubbleSort(0, memory);			
			std::cout << "memory after first sort" << std::endl;
			printMemory(memory, 6);
		}
		index++;
	}	
	R_1 = memory[0];
	R_2 = memory[4];
	product_1 = R_1 * R_2;
	R_1 = memory[1];
	R_2 = memory[3];
	product_2 = R_1 * R_2;
	if (product_1 == product_2)
	{
		R_1 = memory[0];
		R_2 = memory[5];
		product_1 = R_1 * R_2;
		R_1 = memory[2];
		R_2 = memory[3];
		product_2 = R_1 * R_2;
		if (product_1 == product_2)
			std::cout << "these 2 triangles are similar!" << std::endl;
		else
			std::cout << "these 2 triangles are not similar!" << std::endl;
	}
	else
		std::cout << "these 2 triangles are not similar!" << std::endl;
	
	return 0;
}

File: similar-triangles-verification.py
#!/usr/bin/python
# -*- coding: utf-8 -*-


import sys
import getopt
import numpy as np
from datetime import datetime

#===========================================
def main():
#===========================================
    opts, argv = getopt.getopt(sys.argv[1:],'h',['help'])	
    print 'commandline parameters:'
    print 'argv:', type(argv), argv
    print 'opts:', type(opts), opts
    for o in opts:
    	if o in ("-h", "--help"):
	    helpInfo()
                        
    if len(argv) == 0:
    	helpInfo()
	exit()

    elif len(argv) != 6:
	helpInfo()
	exit()
    
    else:
	memory = [int(argv[0]), int(argv[1]), int(argv[2]), int(argv[3]), int(argv[4]), int(argv[5])]

    index = 0
    R_1 = 0
    R_2 = 0
    product_1 = 0
    product_2 = 0

    while index < 2:
        if index == 1:
            bubbleSort(3, memory)
        else:
            bubbleSort(0, memory)
        index = index + 1

    R_1 = memory[0]
    R_2 = memory[4]
    product_1 = R_1 * R_2
    R_1 = memory[1]
    R_2 = memory[3]
    product_2 = R_1 * R_2
    if product_1 == product_2:
	R_1 = memory[0]
	R_2 = memory[5]
	product_1 = R_1 * R_2
	R_1 = memory[2]
	R_2 = memory[3]
	product_2 = R_1 * R_2
        if product_1 == product_2:
            print 'these 2 triangles are similar!'
        else:
	    print 'these 2 triangles are not similar!'
    else:
        print 'these 2 triangles are not similar!'


#===========================================
def bubbleSort(dataIndex, memory):
#===========================================
    R_1 = 0
    R_2 = 0
    flag = True
    while flag == True:
	dataCounter = dataIndex
	flag = False
	while dataCounter < dataIndex + 2:
	    R_1= memory[dataCounter]
	    R_2= memory[dataCounter + 1]
	    if memory[dataCounter] < memory[dataCounter + 1]:
		memory[dataCounter + 1] = R_1
		memory[dataCounter] = R_2
		flag = True
	    dataCounter = dataCounter + 1


#===========================================
def helpInfo():
#===========================================
    print 'input argument error, please input the side lengths of 2 triangles'
    #print 'tplr: calculate total packet loss rate, need parameter <router model> (APU or 150)'

if __name__ == '__main__':
    main()


